diff --git a/gdb/solib-svr4.c b/gdb/solib-svr4.c
index 58432b69a5d..42e5f4dfff1 100644
--- a/gdb/solib-svr4.c
+++ b/gdb/solib-svr4.c
@@ -46,6 +46,7 @@
 #include "probe.h"
 
 #include <map>
+#include "gdbsupport/unordered_map.h"
 
 static void svr4_relocate_main_executable (void);
 static void probes_table_remove_objfile_probes (struct objfile *objfile);
@@ -3766,7 +3767,7 @@ svr4_solib_ops::get_solibs_in_ns (int nsid) const
      faster, and to be able to remove SOs from the map, to avoid
      returning the dynamic linker multiple times.  */
   CORE_ADDR debug_base = info->namespace_id[nsid];
-  std::unordered_map<std::string, const lm_info_svr4 *> namespace_solibs;
+  gdb::unordered_map<std::string, const lm_info_svr4 *> namespace_solibs;
   for (svr4_so &so : info->solib_lists[debug_base])
     namespace_solibs[so.name] = so.lm_info.get ();
 
