diff --git a/gdb/nat/x86-linux.c b/gdb/nat/x86-linux.c
index 16391dcde24..3bf20454e09 100644
--- a/gdb/nat/x86-linux.c
+++ b/gdb/nat/x86-linux.c
@@ -198,6 +198,10 @@ i386_ptrace_get_tls_data (int pid, gdb::array_view<user_desc> buffer)
 {
   gdb_assert (buffer.size () == 3);
 
+#ifndef PTRACE_GET_THREAD_AREA
+#define PTRACE_GET_THREAD_AREA 25
+#endif
+
   for (int i = 0; i < 3; ++i)
     {
       void *addr = (void *) (uintptr_t) (i386_initial_tls_gdt + i);
@@ -217,6 +221,10 @@ i386_ptrace_set_tls_data (int pid, gdb::array_view<user_desc> buffer)
 {
   gdb_assert (buffer.size () == 3);
 
+#ifndef PTRACE_SET_THREAD_AREA
+#define PTRACE_SET_THREAD_AREA 26
+#endif
+
   for (int i = 0; i < 3; ++i)
     {
       void *addr = (void *) (uintptr_t) (i386_initial_tls_gdt + i);
